app/assets/javascripts/components/projects/ProjectBoard.js

```diff
@@ -8,39 +9,112 @@ import Sprints from "../stories/Sprints";
 import History from "../stories/History";
 import { getColumns } from "../../selectors/columns";
 import * as Columns from '../../models/beta/column';
-import { createStory, closeHistory } from '../../actions/story';
+import { createStory, closeHistory, dragDropStory } from '../../actions/story';
 import AddStoryButton from '../story/AddStoryButton';
 import * as Story from 'libs/beta/constants';
 import PropTypes from 'prop-types';
 import { storyPropTypesShape } from '../../models/beta/story';
-import { projectBoardPropTypesShape } from '../../models/beta/projectBoard';
+import {
+  projectBoardPropTypesShape,
+  getNewPosition,
+  getNewSprints,
+  getNewState,
+  moveTask,
+  getBacklogStories,
+  getArray
+} from '../../models/beta/projectBoard';
 import Notifications from '../Notifications';
 import { removeNotification } from '../../actions/notifications';
 import StorySearch from '../search/StorySearch';
 import SearchResults from './../search/SearchResults';
 import ProjectLoading from './ProjectLoading';
 
-export class ProjectBoard extends React.Component {
-  componentWillMount() {
-    this.props.fetchProjectBoard(this.props.projectId);
-  }
+const ProjectBoard = ({
+  projectId,
+  createStory,
+  closeHistory,
+  notifications,
+  removeNotification,
+  history,
+  projectBoard,
+  fetchProjectBoard,
+  dragDropStory,
+  chillyBinStories,
+  backlogSprints,
+  doneSprints,
+  fetchPastStories
+}) => {
+  const [newChillyBinStories, setNewChillyBinStories] = useState();
+  const [newBacklogSprints, setNewBacklogSprints] = useState();
+
+  useEffect(() => {
+    setNewBacklogSprints(backlogSprints);
+  }, [backlogSprints]);
+
+  useEffect(() => {
+    setNewChillyBinStories(chillyBinStories);
+  }, [chillyBinStories]);
+
+  useEffect(() => {
+    fetchProjectBoard(projectId);
+  }, []);
```

Comment:
Its a good pratice to include all dependecies in the useEffect dependencies array:
```js
 useEffect(() => {
    fetchProjectBoard(projectId);
  }, [fetchProjectBoard, projectId]);
```